taboola.module

```diff
@@ -288,10 +289,16 @@ function taboola_get_widget_setting_js($block_index) {
 
   // Get target type setting.
   $target_type = taboola_get_block_setting($block_index, 'target_type');
-  $target_type_json = !empty($target_type) ? ", target_type:'" . check_plain($target_type) . "'" : '';
+
+  $widget_settings = array();
+  $widget_settings['mode'] = $mode;
+  $widget_settings['container'] = $container;
+  $widget_settings['placement'] = $placement;
+  $widget_settings['target_type'] = !empty($target_type) ?  $target_type : '';
+  $widget_settings_json = json_encode($widget_settings);
```

Comment:
Perhaps we should use drupal_json_encode() here?

Reply:
... it adds `JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT` options